packages/SwingSet/src/vats/comms/state.js

```diff
@@ -704,6 +704,7 @@ export function makeState(syscall) {
     store.set(`${lpid}.data.slots`, data.slots.join(','));
     let idx = 0;
     for (const slot of data.slots) {
+      // @ts-expect-error may not be string
       incrementRefCount(slot, `resolve|s${idx}`);
```

Comment:
The kernel only ever uses strings, although in some places they're krefs and in others they're vrefs, (and comms also has rrefs and lrefs). I bet it'd catch some potentials bugs if we had distinct types for those.

I think `CapData` is actually defined by `marshal`, which doesn't enforce `typeof(slot)==='string'`, so `CapData` is a generic (parameterized by some type T, which the kernel always fills with `string`). I think we have a more-specialized `KernelCapData` or `SwingsetCapData` within `packages/SwingSet/`, since we can't use the generic one.

I can't think of a current use case for non-string slots, so I'd probably be ok if `marshal` went ahead and assumed they were always strings. But I also expect that the `marshal` folks (@erights) would think of that as unnecessarily narrow / overly constrained.

Reply:
Thanks for explaining that. A quick fix was to update the `insistCapData` assertion.